backend/plugin/db/hive/dump.go

```diff
@@ -2,16 +2,229 @@ package hive
+				)
+				// TODO(tommy): get more index info from SyncInstance.
+				indexDDL, err := genIndexDDL(&IndexDDLOptions{
+					indexName:             index.Name,
+					tableName:             table.Name,
+					indexType:             index.Type,
+					colNames:              index.Expressions,
+					isWithDeferredRebuild: true,
+					idxProperties:         nil,
+					indexTableName:        "",
+					rowFmt:                "",
+					storedAs:              "",
+					storedBy:              "",
+					location:              "",
+					tableProperties:       nil,
+					comment:               index.Comment,
+				})
+				if err != nil {
+					return "", errors.Wrapf(err, "failed to generate DDL for index %s", index.Name)
+				}
+
+				_, _ = dumpStrBuilder.WriteString(fmt.Sprintf(schemaStmtFmt, "INDEX", tabNameWithDB, indexDDL))
+			}
+			_, _ = dumpStrBuilder.WriteString(tabDDL)
+		}
+
+		// dump external tables.
+		for _, extTable := range database.ExternalTables {
+			tabDDL, err := d.showCreateSchemaDDL(ctx, "TABLE", extTable.Name, database.Name)
+			if err != nil {
+				return "", errors.Wrapf(err, "failed to dump table %s", extTable.Name)
+			}
+			_, _ = dumpStrBuilder.WriteString(tabDDL)
+		}
+
+		// dump views.
+		for _, view := range database.Views {
+			viewDDL, err := d.showCreateSchemaDDL(ctx, "VIEW", view.Name, database.Name)
+			if err != nil {
+				return "", errors.Wrapf(err, "failed to dump view %s", view.Name)
+			}
+			_, _ = dumpStrBuilder.WriteString(viewDDL)
+		}
+
+		// dump materialized views.
+		for _, mtView := range database.MaterializedViews {
+			mtViewDDL := fmt.Sprintf(mtViewDDLFormat, mtView.Name, mtView.Definition)
+			_, _ = dumpStrBuilder.WriteString(fmt.Sprintf(schemaStmtFmt, "MATERIALIZED VIEW", mtView.Name, mtViewDDL))
+		}
+	}
+
+	return dumpStrBuilder.String(), nil
 }
 
 // Restore the database from src, which is a full backup.
 func (*Driver) Restore(_ context.Context, _ io.Reader) error {
 	return errors.Errorf("Not implemeted")
 }
+
+// This function shows DDLs for creating certain type of schema [VIEW, DATABASE, TABLE].
+func (d *Driver) showCreateSchemaDDL(ctx context.Context, schemaType string, schemaName string, belongTo string) (string, error) {
+	// 'SHOW CREATE TABLE' can also be used for dumping views.
+	queryStatement := fmt.Sprintf("SHOW CREATE %s `%s`", schemaType, schemaName)
+	if schemaType == "VIEW" {
+		queryStatement = fmt.Sprintf("SHOW CREATE TABLE `%s`", schemaName)
+	}
+
+	schemaDDLResults, err := d.QueryConn(ctx, nil, queryStatement, nil)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to dump %s %s", schemaType, schemaName)
+	}
+
+	var schemaDDL string
+	for _, row := range schemaDDLResults[0].Rows {
+		schemaDDL += fmt.Sprintln(row.Values[0].GetStringValue())
+	}
+
+	// rename table to format: [DatabaseName].[TableName].
+	newSchemaName := schemaName
+	if belongTo != "" {
+		newSchemaName = fmt.Sprintf("%s.%s", belongTo, schemaName)
```

Comment:
Is there need back-quoted?

Reply:
some back-quotes are added in somewhere else, which is kinda messy. fix it later~